src/Compilers/Core/RebuildTest/OptionRoundTripTests.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System;
+using System.Reflection.PortableExecutable;
+using BuildValidator;
+using Castle.Core.Logging;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Emit;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.Extensions.Logging;
+using Xunit;
+using System.IO;
+using System.Threading;
+using Microsoft.CodeAnalysis.VisualBasic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.Rebuild.UnitTests
+{
+    public class OptionRoundTripTests : CSharpTestBase
+    {
+        public static readonly object[][] s_platforms = ((Platform[])Enum.GetValues(typeof(Platform))).Select(p => new[] { (object)p }).ToArray();
+
+        [Theory]
+        [MemberData(nameof(s_platforms))]
+        public void Platform_RoundTrip(Platform platform)
+        {
+            const string path = "test";
+
+            var original = CreateCompilation(
+                "class C { static void Main() { } }",
+                options: TestOptions.DebugExe.WithPlatform(platform));
+
+            original.VerifyDiagnostics();
+
+            var originalBytes = original.EmitToArray(new EmitOptions(debugInformationFormat: DebugInformationFormat.Embedded));
+            var peReader = new PEReader(originalBytes);
+            Assert.True(peReader.TryOpenAssociatedPortablePdb(path, path => null, out var provider, out _));
+            var pdbReader = provider!.GetMetadataReader();
+
+            var factory = LoggerFactory.Create(configure => { });
+            var logger = factory.CreateLogger(path);
+            // TODO: shouldn't need to pass a logger.
```

Comment:
If we need an `ILogger` instance, could we create a simple implementation of `ILogger` in this test clas?

Reply:
Yes, but it feels like users of the public API shouldn't be required to provide a logger, and shouldn't provide it as its own parameter. Maybe a property on an options object that can be specified if the user wants, but not a parameter to public API that they have to deal with.